### DISCUSSION TITLE
Panel to display the GPS recording time

### DISCUSSION THREAD

**Discussion by printmaps** [devemux86/cruiser#660] [Q&A] [answered]
Currently there is no panel for the trip time. Please add one.

This is meant with trip time:

- start a timer when track recording is started
- update the timer during track recording
- stop the time when track recording is stopped

<img src="https://github.com/user-attachments/assets/44953222-e434-4c0d-bb6b-d8e2566a3bb5" width="50%"/><img src="https://github.com/user-attachments/assets/92ff59d0-d555-42c5-a850-a4ce01bd1e05" width="50%"/>

**Comment by devemux86** [comment]
@printmaps Thanks for the suggestion.

We already have 2 separate panels for the traveled distance. We could also add for the traveled time.

Note that the distance panels work automatically in navigation or follow mode.
There is no manual start / stop. Just reset and set actions.

**Comment by printmaps** [comment]
> We could also add for the traveled time.

That sounds good.

**Comment by Klaus-Tockloth** [comment]
The idea was as follows:

- start GPS recording
- start of time measurement

The time measurement refers to the start time of the GPS recording.

Correspondingly:
- stop GPS recording
- stop time measurement

The planned implementation refers to the trip odometer. At the moment I am not sure whether the above
desired will be achieved.

**Comment by devemux86** [comment]
We can add a menu (via long press) to the GPS recording panel that now shows the recorded distance, for what to show:
- Recorded distance
- Recorded time

**Reply by devemux86** [reply]
Or display the recorded distance and time simultaneously in 2 panels?

**Comment by printmaps** [comment]
> Or display the recorded distance and time simultaneously in 2 panels?

I vote for the above.

**Comment by devemux86** [comment]
Implementation with 2 (optional) GPS recording panels for distance / time:

<img src="https://github.com/user-attachments/assets/292d91e2-9340-4827-a289-be1f61ecb60a" width="50%"/><img src="https://github.com/user-attachments/assets/1ca90426-92b9-4c94-9a99-2107482dcd92" width="50%"/>

**Reply by bios4** [reply]
Suggestion:

<img src="https://github.com/user-attachments/assets/5bad6925-07e3-4320-8ce3-067f192ed9b1" width="50%">

**Reply by bios4** [reply]
Plus: I am really NOT happy to have TWO start/stop recording buttons now!
But let's see how this turns out in reality...

Can all of these "second" options be simply (fully) deactivated?

**Reply by devemux86** [reply]
Everything is optional as usual, the settings are shown in the 2nd image.

---

Travel icons are best discussed in their thread, also read their description where the "filled" icon is explained:
- https://github.com/devemux86/cruiser/discussions/670

**Reply by DolomitesTom** [reply]
what is the benefit to have this separated? And what a about a former request of mine to have the average speed of a trip with and without pauses (2 separate infos) in a panel and also in the track details?

**Reply by devemux86** [reply]
2 values cannot be displayed in 1 panel.

---

About average speed, it is measured when device moves and speed is reported by GPS.
By calculating the average of a value, time is not needed.

**Comment by printmaps** [comment]
The implementation looks good and is consistent in relation to the two trip counters.

**Comment by devemux86** [comment]
**[Cruiser 5.1.0-b2](https://github.com/devemux86/cruiser/discussions/671)** displays zeros in these panels too when there is no recorded, so that we can easily separate them.

**Comment by devemux86** [comment, marked as answer]
- New panel to display the GPS recording time is available in https://github.com/devemux86/cruiser/discussions/684
